### PR TITLE
chore: example generated instrumentation

### PR DESCRIPTION
**Pull request by robertlaurin**
For demonstration of https://github.com/open-telemetry/opentelemetry-ruby/pull/496 only.

### CODE REVIEW

File: instrumentation/amazing_gem/.yardopts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+--no-private
+--title=OpenTelemetry amazing_gem Instrumentation
```
Comment (fbogsany) [inline instrumentation/amazing_gem/.yardopts:2]: We've generally turned this into capitalized words, e.g.
```
concurrent_ruby => Concurrent Ruby
delayed_job => Delayed Job
dalli => Dalli
```

File: instrumentation/amazing_gem/Gemfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
```
Comment (fbogsany) [inline instrumentation/amazing_gem/Gemfile:3]: The [preferred copyright notice](https://github.com/cncf/foundation/blob/master/copyright-notices.md) is:
```ruby
# Copyright The OpenTelemetry Authors
```
I.e. we shouldn't include the year.

Comment (robertlaurin) [reply instrumentation/amazing_gem/Gemfile:3]: Should the license be updated as well? 

https://github.com/open-telemetry/opentelemetry-ruby/pull/497/files#diff-83b5e236433acba18ca2306b043b5c196bf7bf2c5eab692dbd90eabc0c20d4f1R189

Comment (fbogsany) [reply instrumentation/amazing_gem/Gemfile]: Good catch! Yes, please.

File: .toys/.data/releases.yml [hunk 3]
```diff
@@ -31,6 +31,10 @@ commit_lint:
 #  *  changelog_path: Path to CHANGLEOG.md relative to the gem directory.
 #     (Required only if it is not in the expected location.)
 gems:
+  - name: opentelemetry-instrumentation-amazing_gem
+    directory: instrumentation/amazing_gem
+    version_constant: [OpenTelemetry, Instrumentation, AmazingGem, VERSION]
```
Comment (fbogsany) [inline .toys/.data/releases.yml:36]: 🎉

File: instrumentation/all/opentelemetry-instrumentation-all.gemspec [hunk 4]
```diff
@@ -26,6 +26,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ['lib']
   spec.required_ruby_version = '>= 2.5.0'
 
+  spec.add_dependency 'opentelemetry-instrumentation-amazing_gem', '~> 0.10.0'
```
Comment (fbogsany) [inline instrumentation/all/opentelemetry-instrumentation-all.gemspec:29]: This doesn't match the `VERSION` in `version.rb`.

Comment (robertlaurin) [reply instrumentation/all/opentelemetry-instrumentation-all.gemspec:29]: Fixed.

**Comment by robertlaurin** [comment]
Failures are due to rubocop not being happy about the lack of alphabetical ordering. 

```
Gemfile:12:1: C: Bundler/OrderedGems: Gems should be sorted in an alphabetical order within their section of the Gemfile. Gem opentelemetry-api should appear before opentelemetry-instrumentation-amazing_gem.
gem 'opentelemetry-api', path: '../../api'
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
opentelemetry-instrumentation-all.gemspec:30:3: C: Gemspec/OrderedDependencies: Dependencies should be sorted in an alphabetical order within their section of the gemspec. Dependency opentelemetry-instrumentation-active_model_serializers should appear before opentelemetry-instrumentation-amazing_gem.
  spec.add_dependency 'opentelemetry-instrumentation-active_model_serializers', '~> 0.10.0'
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I guess I could add a generator step that runs rubocop to fixup any issues in the newly created gem dir...

**Review by fbogsany — COMMENTED** [review]
generated code looks good, thanks @robertlaurin !

**Pull request closed without merging**